docs/sources/logql/query_examples.md

```diff
@@ -62,3 +64,136 @@ and only include errors whose duration is above ten seconds.
         | json
         | duration > 10s [1m]))
     ```
+
+## Multiple filtering stages examples
+
+Query results are gathered by successive evaluation of parts of the query from left to right.
+To make querying efficient,
+order the filtering stages left to right:
+
+1. stream selector
+2. line filters
+3. label filters
+
+Consider the query:
+
+```logql
+{cluster="ops-tools1", namespace="loki-dev", job="loki-dev/query-frontend"} |= "metrics.go" != "out of order" | logfmt | duration > 30s or status_code != "200"
+```
+Within this query, the stream selector is
+
+```
+{cluster="ops-tools1", namespace="loki-dev", job="loki-dev/query-frontend"}
+```
+
+There are two line filters: 
+`|= "metrics.go"`
+and `!="out of order"`.
+Of the log lines identified with the stream selector,
+the query results
+include only those log lines that contain the string "metrics.go"
+and do not contain the string "out of order".
+
```

Comment:
It is missing the explanation of the `| logfmt` parser 👀 
In order to have access to labels `duration` and `status_code` this parser needs to be applied.